src/Tasks/ResourceHandling/MSBuildResXReader.cs

```diff
@@ -0,0 +1,300 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Shared;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace Microsoft.Build.Tasks.ResourceHandling
+{
+    internal class MSBuildResXReader
+    {
+        public static IReadOnlyList<IResource> ReadResources(Stream s, string filename, bool pathsRelativeToBasePath)
+        {
+            var resources = new List<IResource>();
+            var aliases = new Dictionary<string, string>();
+
+            using (var xmlReader = new XmlTextReader(s))
+            {
+                xmlReader.WhitespaceHandling = WhitespaceHandling.None;
+
+                XDocument doc = XDocument.Load(xmlReader, LoadOptions.PreserveWhitespace);
+                foreach (XElement elem in doc.Element("root").Elements())
+                {
+                    switch (elem.Name.LocalName)
+                    {
+                        case "assembly":
+                            ParseAssemblyAlias(aliases, elem);
+                            break;
+                        case "resheader":
+                            break;
+                        case "data":
+                            ParseData(filename, pathsRelativeToBasePath, resources, aliases, elem);
```

Comment:
nit: make ParseData return `resources`, to make it clear what it's final goal is.